src/main/frontend/extensions/code.cljs

```diff
@@ -318,13 +317,13 @@
   ;; you're trying to focus doesn't yet exist. Adding the requestAnimationFrame
   ;; ensures that the React component re-renders before the :codemirror/focus
   ;; command is run. It's not elegant... open to suggestions for how to fix it!
-  (js/window.requestAnimationFrame
-   (fn []
-     (let [block (state/get-edit-block)
-           block-uuid (:block/uuid block)
-           block-node (util/get-first-block-by-id block-uuid)]
-       (editor-handler/select-block! (:block/uuid block))
-       (let [textarea-ref (.querySelector block-node "textarea")]
-         (.focus (gobj/get textarea-ref codemirror-ref-name)))
-       (util/select-unhighlight! (dom/by-class "selected"))
-       (state/clear-selection!)))))
+  (let [block (state/get-edit-block)
+        block-uuid (:block/uuid block)]
+    (state/clear-edit!)
+    (js/setTimeout
+     (fn []
+       (let [block-node (util/get-first-block-by-id block-uuid)
+             textarea-ref (.querySelector block-node "textarea")]
+         (when-let [codemirror-ref (gobj/get textarea-ref codemirror-ref-name)]
+           (.focus codemirror-ref))))
+     0)))
```

Comment:
need a >100 timeout on Safari, or the dom "textarea" won't be ready.
```
(.querySelector block-node "textarea") ;;=> nil if timeout=0
```

Reply:
Thanks, updated!